src/ome_zarr_models/v04/coordinate_transformations.py

```diff
@@ -69,3 +92,32 @@ class PathTranslation(Base):
 
     type: Literal["translation"]
     translation: str
+
+
+def ndim(transform: VectorScale | VectorTranslation) -> int:
```

Comment:
In the spirit of don't repeat yourself, I'm not mad about having `ndim` defined as both a method on the classes and a function. Was there a particular reason for doing this?

Reply:
the main reason to have a free-floating function is to support the niche case of using a functional idiom like `map` or `filter` on transformation objects, e.g. `ndims = map(ndim, transforms)`, but maybe this is too niche :)